File: apps/api/src/application/imports/list-import-jobs.ts
import type { ImportJobRepository } from '@/application/ports/import-job-repository'
import type { ImportJobSummary } from '@/domain/import/import-job'

export class ListImportJobs {
  constructor(private readonly imports: ImportJobRepository) {}

  execute(): Promise<ImportJobSummary[]> {
    return this.imports.findAll()
  }
}

File: apps/api/src/ecommerce_api/application/imports/list_import_jobs.py
from ecommerce_api.application.ports.import_job_repository import ImportJobRepository
from ecommerce_api.domain.import_job import ImportJobSummary


class ListImportJobs:
    def __init__(self, imports: ImportJobRepository) -> None:
        self._imports = imports

    async def execute(self) -> list[ImportJobSummary]:
        return await self._imports.find_all()
